crates/bevy_render/src/texture/image.rs

```diff
@@ -687,6 +688,364 @@ impl Image {
+        let len = self.texture_descriptor.format.pixel_size();
+        self.pixel_data_offset(coords)
+            .map(|start| &mut self.data[start..(start + len)])
+    }
+
+    /// Read the color of a specific pixel.
+    ///
+    /// This function will find the raw byte data of a specific pixel and
+    /// decode it into a user-friendly [`Color`] struct for you.
+    ///
+    /// Supports many of the common [`TextureFormat`]s:
+    ///  - RGBA/BGRA 8-bit unsigned integer, both sRGB and Linear
+    ///  - 16-bit and 32-bit unsigned integer
+    ///  - 32-bit float
+    ///
+    /// Be careful: as the data is converted to [`Color`] (which uses `f32` internally),
+    /// there may be issues with precision when using non-float [`TextureFormat`]s.
+    /// If you read a value you previously wrote using `set_color_at`, it will not match.
+    /// If you are working with a 32-bit integer [`TextureFormat`], the value will be
+    /// inaccurate (as `f32` does not have enough bits to represent it exactly).
+    ///
+    /// Single channel (R) formats are assumed to represent greyscale, so the value
+    /// will be copied to all three RGB channels in the resulting [`Color`].
+    ///
+    /// Other [`TextureFormat`]s are unsupported, such as:
+    ///  - block-compressed formats
+    ///  - non-byte-aligned formats like 10-bit
+    ///  - 16-bit float formats
+    ///  - signed integer formats
+    #[inline(always)]
+    pub fn get_color_at(&self, coords: UVec3) -> Result<Color, TextureAccessError> {
+        let Some(bytes) = self.pixel_bytes(coords) else {
+            return Err(TextureAccessError::OutOfBounds {
+                x: coords.x,
+                y: coords.y,
+                z: coords.z,
+            });
+        };
+
+        match self.texture_descriptor.format {
+            TextureFormat::Rgba8UnormSrgb => Ok(Color::rgba(
+                bytes[0] as f32 / u8::MAX as f32,
+                bytes[1] as f32 / u8::MAX as f32,
+                bytes[2] as f32 / u8::MAX as f32,
+                bytes[3] as f32 / u8::MAX as f32,
+            )),
+            TextureFormat::Rgba8Unorm | TextureFormat::Rgba8Uint => Ok(Color::rgba_linear(
+                bytes[0] as f32 / u8::MAX as f32,
+                bytes[1] as f32 / u8::MAX as f32,
+                bytes[2] as f32 / u8::MAX as f32,
+                bytes[3] as f32 / u8::MAX as f32,
+            )),
+            TextureFormat::Bgra8UnormSrgb => Ok(Color::rgba(
+                bytes[3] as f32 / u8::MAX as f32,
+                bytes[2] as f32 / u8::MAX as f32,
+                bytes[1] as f32 / u8::MAX as f32,
+                bytes[0] as f32 / u8::MAX as f32,
+            )),
+            TextureFormat::Bgra8Unorm => Ok(Color::rgba_linear(
+                bytes[3] as f32 / u8::MAX as f32,
+                bytes[2] as f32 / u8::MAX as f32,
+                bytes[1] as f32 / u8::MAX as f32,
+                bytes[0] as f32 / u8::MAX as f32,
```

Comment:
```suggestion
                bytes[2] as f32 / u8::MAX as f32,
                bytes[1] as f32 / u8::MAX as f32,
                bytes[0] as f32 / u8::MAX as f32,
                bytes[3] as f32 / u8::MAX as f32,
```

Reply:
Oh correct, good catch!

Such code is very mechanical and it is easy to menally zone out when writing it and make mistakes. It's very boring code, but tedious to verify everything is correct. Thanks for reviewing it!